pkg/cgroups/memory.go

```diff
@@ -45,16 +44,25 @@ func (c *memHandler) Stat(ctr *CgroupControl, m *Metrics) error {
 
 	if ctr.cgroup2 {
 		memoryRoot = filepath.Join(cgroupRoot, ctr.path)
-		filenames["usage"] = "memory.current"
+		filenames["usage"] = "memory.stat"
 		filenames["limit"] = "memory.max"
 	} else {
 		memoryRoot = ctr.getCgroupv1Path(Memory)
 		filenames["usage"] = "memory.usage_in_bytes"
 		filenames["limit"] = "memory.limit_in_bytes"
 	}
-	usage.Usage, err = readFileAsUint64(filepath.Join(memoryRoot, filenames["usage"]))
-	if err != nil {
-		return err
+
+	if filenames["usage"] == "memory.stat" {
+		anon, err := readFileByKeyAsUint64(filepath.Join(memoryRoot, filenames["usage"]), "anon")
```

Comment:
nit: why are you using a temporary variable here, but in the else block define it directly?

usage.Usage, err := ...

Then you can move the if err ... Block to below the if then else.

Reply:
sorry, my bad, the first trial of this PR is a `sum` from `active_anon` AND `active_anon`,  then I found that use `anon` is better, but forget to refactor the code -_-

Reply:
updated and force pushed.